gson/src/main/resources/META-INF/proguard/gson.pro

```diff
@@ -59,13 +54,16 @@
   <init>();
 }
 
-# If a class is used in some way by the application and has fields annotated with @SerializedName,
-# keep those fields and the constructors of the class
-# For convenience this also matches classes without no-args constructor, in which case Gson uses JDK Unsafe
-# Based on https://issuetracker.google.com/issues/150189783#comment11
-# See also https://github.com/google/gson/pull/2420#discussion_r1241813541 for a more detailed explanation
+# Keep fields annotated with @SerializedName for classes which are present.
+# If classes with fields annotated with @SerializedName have a no-args
+# constructor keep that as well.
```

Comment:
Why did you remove the link to the pull request discussion?

At least in my opinion this `-if class * ...` is pretty cryptic and its intended behavior is not directly obvious, so it would be good to have a reference explaining what this actually does.

Reply:
Good point, added those references back.